examples/volumes/vsphere/vsphere-volume-sc-vsancapabilities.yaml

```diff
@@ -0,0 +1,9 @@
+kind: StorageClass
+apiVersion: storage.k8s.io/v1beta1
+metadata:
+  name: fast
+provisioner: kubernetes.io/vsphere-volume
+parameters:
+    diskformat: zeroedthick
+    hostFailuresToTolerate: "2"
```

Comment:
Agree with Sandeep. For policy based provisioning, user expect system to determine datastore for him.

If the datastore is not specified in the storage class yaml and default datastore in vsphere.conf (vs.cfg.Global.Datastore) is set to some shared vmfs/nfs datastore, In this case we are simply failing create call.

If volumeOptions.StorageProfileData != "" then let’s find the compatible vsan datastore and then continue with provisioning. 

If user has already specified some VSAN datastore, then we should check, if this VSAN datastore comply to the policy, he has specified.

In case of multiple vsan datastore present in the VC, we should also check if user preferred vsan datastore is accessible to all nodes, otherwise later we will not be able to attach these disks to any node.